airflow/contrib/hooks/azure_batchai_hook.py

```diff
@@ -0,0 +1,111 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+
+from airflow.hooks.base_hook import BaseHook
+from airflow.exceptions import AirflowException
+
+from azure.common.client_factory import get_client_from_auth_file
+from azure.common.credentials import ServicePrincipalCredentials
+
+from azure.mgmt.batchai import BatchAIManagementClient
+
+
+class AzureBatchAIHook(BaseHook):
+    """
+    Interact with Azure Batch AI
+    :param azure_batchai_conn_id: Reference to the Azure Batch AI connection
+    :type azure_batchai_conn_id: str
+    :param config_data: JSON Object with credential and subscription information
+    :type config_data: str
+    """
+
+    def __init__(self, azure_batchai_conn_id='azure_batchai_default', config_data=None):
+        self.conn_id = azure_batchai_conn_id
+        self.connection = self.get_conn()
```

Comment:
Why we need this? An object should be created when it is needed.

Reply:
Do you recommend I initialize it as None? I refer to self.connection later in the code so I didn't want it to not be defined

Reply:
Yes. I think it will be better.